### PR TITLE
SearchV2: Use the same components for both SQL and bluge backends

### PR DESCRIPTION
**Pull request by ryantxu**
This updates the UI so we use the same components when the `panelTitleSearch` feature flag is enabled or not.  This allows us to remove LOTS of old/complicated/internal ID based components and logic.  The best kind of PR:
image

This PR:
1. Always uses the new components
2. Removes many of the old search components and redux options

Note, we can flip between the two search implementations using:
* SQL = http://localhost:3000/dashboards?__feature.panelTitleSearch=false
* Bluge = http://localhost:3000/dashboards?__feature.panelTitleSearch=true

### CODE REVIEW

File: public/app/features/search/components/DashboardListPage.tsx [hunk 1]
```diff
@@ -46,22 +45,16 @@ export const DashboardListPage: FC<Props> = memo(({ navModel, match, location })
...
-      )}
+      <Page.Contents
+        isLoading={loading}
+        className={css`
+          display: flex;
+          flex-direction: column;
+          overflow: hidden;
+        `}
+      >
+        <ManageDashboardsNew folder={value?.folder} />
```
Comment (ryantxu) [inline public/app/features/search/components/DashboardListPage.tsx:56]: keeping this as `...New` for now, just to make git not show the diff between old vs new

**Review by axelavargas — COMMENTED** [review]
🤔 I found some weird issue, but I am not sure if it is related to this PR. 

When I click on "Go to Folder" sometimes the query request fails and does not show results

https://user-images.githubusercontent.com/239999/176456133-4e663719-dea1-420c-abbe-76f3c4c42093.mp4

**Comment by ryantxu** [comment]
Interesting, the"CRC check failed" error is from the `/ds/query` endpoint -- and we need to figure out why you get it.  But unrelated to this change where the requests still hit:  /api/search

**Review by axelavargas — APPROVED** [review]
Hey @ryantxu , last tests, and works as expected with and without feature flag 🚀 

I checked again with a clean environment and the error I reported is already in the main branch and unrelated to this PR. 

🥳

**Pull request merged**